test/e2e/pods.go

```diff
@@ -555,6 +578,233 @@ var _ = Describe("Pods", func() {
+		}
+
+		By("verifying the pod is in kubernetes")
+		pod, err = podClient.Get(name)
+		if err != nil {
+			Failf("failed to get pod: %v", err)
+		}
+
+		By("updating the image")
+		pod.Spec.Containers[0].Image = "nginx"
+		pod, err = podClient.Update(pod)
+		if err != nil {
+			Failf("error updating pod=%s/%s %v", name, containerName, err)
+		}
+		time.Sleep(10 * time.Second)
+		expectNoError(framework.WaitForPodRunning(pod.Name))
+
+		By("get restart delay after image update")
+		delayAfterUpdate, err := getRestartDelay(framework.Client, pod, framework.Namespace.Name, name, containerName)
+		if err != nil {
+			Failf("timed out waiting for container restart in pod=%s/%s", name, containerName)
+		}
+
+		if delayAfterUpdate > delay2 || delayAfterUpdate > delay1 {
+			Failf("updating image did not reset the back-off value in pod=%s/%s d3=%s d2=%s d1=%s", name, containerName, delayAfterUpdate, delay1, delay2)
+		}
+
+	})
+
+	It("should not back-off on LivenessProbe failure", func() {
+		podClient := framework.Client.Pods(framework.Namespace.Name)
+		name := "back-off-liveness-exec"
+		containerName := "liveness"
+		pod := &api.Pod{
+			ObjectMeta: api.ObjectMeta{
+				Name:   name,
+				Labels: map[string]string{"test": "liveness"},
+			},
+			Spec: api.PodSpec{
+				Containers: []api.Container{
+					{
+						Name:    containerName,
+						Image:   "gcr.io/google_containers/busybox",
+						Command: []string{"/bin/sh", "-c", "echo ok >/tmp/health; sleep 5; rm -rf /tmp/health; sleep 600"},
+						LivenessProbe: &api.Probe{
+							Handler: api.Handler{
+								Exec: &api.ExecAction{
+									Command: []string{"cat", "/tmp/health"},
+								},
+							},
+							InitialDelaySeconds: 2,
+						},
+					},
+				},
+			},
+		}
+
+		By("submitting the pod to kubernetes")
+		defer func() {
+			By("deleting the pod")
+			podClient.Delete(pod.Name, api.NewDeleteOptions(0))
+		}()
+		pod, err := podClient.Create(pod)
+		if err != nil {
+			Failf("Failed to create pod: %v", err)
+		}
+
+		expectNoError(framework.WaitForPodRunning(pod.Name))
+
+		By("verifying the pod is in kubernetes")
+		pod, err = podClient.Get(name)
+		if err != nil {
+			Failf("failed to get pod: %v", err)
+		}
+
+		By("geting restart delay-1")
+		delay1, err := getRestartDelay(framework.Client, pod, framework.Namespace.Name, name, containerName)
+		if err != nil {
+			Failf("timed out waiting for container restart in pod=%s/%s", name, containerName)
+		}
+
+		By("getting restart delay-2")
+		delay2, err := getRestartDelay(framework.Client, pod, framework.Namespace.Name, name, containerName)
+		if err != nil {
+			Failf("timed out waiting for container restart in pod=%s/%s", name, containerName)
+		}
+
+		ratio := float64(delay2) / float64(delay1)
+		if math.Floor(ratio) != 1 && math.Ceil(ratio) != 1 {
+			Failf("back-off increasing on LivenessProbe failure delay1=%s delay2=%s", delay1, delay2)
+		}
+	})
+
+	It("[Skipped] should cap back-off at 5mins", func() {
```

Comment:
Where does the "5 minutes" amount come from?  Does it need to correspond to some other constant somewhere?